src/gt4py/next/iterator/transforms/global_tmps.py

```diff
@@ -485,10 +485,19 @@ def update_domains(node: FencilWithTemporaries, offset_provider: Mapping[str, An
                         new_axis = nbt_provider.neighbor_axis.value
                         consumed_domain.ranges.pop(old_axis)
                         assert new_axis not in consumed_domain.ranges
-                        consumed_domain.ranges[new_axis] = SymbolicRange(
-                            im.literal("0", ir.INTEGER_INDEX_BUILTIN),
-                            im.literal(str(horizontal_sizes[new_axis]), ir.INTEGER_INDEX_BUILTIN),
-                        )
+
+                        if symbolic_sizes is not None:
```

Comment:
For readability the if condition should not be negated without good reason when there is an else clause.